routers/web/repo/activity.go

```diff
@@ -52,6 +53,12 @@ func Activity(ctx *context.Context) {
 	ctx.Data["DateUntil"] = timeUntil
 	ctx.Data["PeriodText"] = ctx.Tr("repo.activity.period." + ctx.Data["Period"].(string))
 
+	branchExist, _ := git.IsBranchExist(ctx, ctx.Repo.Repository.ID, ctx.Repo.Repository.DefaultBranch)
```

Comment:
Since this page include code, issues and other information. Maybe just consider not return 404 but use `ctx.Repo.CanRead(unit.TypeCode) & branchExist` as the parameter.

Reply:
https://github.com/go-gitea/gitea/pull/34096/commits/7c5d7af5daf4f50a44010b0e8b7b9002a2f7a6b1